chia/wallet/puzzles/nft_metadata_updater_default.clvm

```diff
@@ -1,15 +1,29 @@
 (mod (CURRENT_METADATA METADATA_UPDATER_PUZZLE_HASH solution)
```

Comment:
it's good progress, but still quite complex and not sure why we need to support multiple field updating at once. 
Here's code that's much simpler and only does one iteration over METADATA: 
```suggestion
(mod (CURRENT_METADATA METADATA_UPDATER_PUZZLE_HASH (key new_url))

  ; METADATA and METADATA_UPDATER_PUZZLE_HASH are passed in as truths from the layer above

  ; This program returns ((new_metadata new_metadata_updater_puzhash) conditions)

  ; once we find the key we don't need to continue looping

  (defun add_url (METADATA key new_url)
    (if METADATA
      (if (= (f (f METADATA)) key)
        (c (c key (c new_url (r (f METADATA)))) (r METADATA))
        (c (f METADATA) (add_url (r METADATA) key new_url))
      )
      ()
    )
  )

  ; main
  ; returns ((new_metadata new_metadata_updater_puzhash) conditions)
  (list
    (list
        (if (all key new_url)
            (if (any (= key "mu") (= key "lu") (= key "u"))
                (add_url CURRENT_METADATA key new_url)
                CURRENT_METADATA
            )
            CURRENT_METADATA
         )
        METADATA_UPDATER_PUZZLE_HASH)
    0
  )
)
```

Reply:
For the users, they may not want to spend the NFT multiple times if they want to update multiple fields, kind of expensive.

Reply:
You're assuming that most common scenario is that all 3 links will become obsolete at once and need to be changed, and I'd say it's much more likely only one of them will become invalid and require another one. Because in current form it's much more expensive adding a single url, right?